tests/test_smartcar.py

```diff
@@ -189,7 +189,56 @@ def test_get_auth_url_vehicle_info_dictionary(self):
         })
 
         expected = smartcar.const.CONNECT_URL + '/oauth/authorize?' + query
-        
+
+        expected_params = parse_qs(expected)
+        actual_params = parse_qs(actual)
+
+        assertDeepEquals(self, expected_params, actual_params)
+
+    def test_get_auth_url_single_select_bool(self):
+        client = smartcar.AuthClient(self.client_id, self.client_secret,
+                self.redirect_uri, self.scope, development=False)
+
+        actual = client.get_auth_url(force=True, state='stuff', single_select=True)
+
+        query = urlencode({
+            'response_type': 'code',
+            'client_id': self.client_id,
+            'redirect_uri': self.redirect_uri,
+            'approval_prompt': 'force',
+            'state': 'stuff',
+            'scope': ' '.join(self.scope),
+            'single_select': True
+        })
+
+        expected = smartcar.const.CONNECT_URL + '/oauth/authorize?' + query
+
+        expected_params = parse_qs(expected)
+        actual_params = parse_qs(actual)
+
+        assertDeepEquals(self, expected_params, actual_params)
+    def test_get_auth_url_single_select_dictionary_vin(self):
+        single_select = {
+            'vin': '12345678901234'
+        }
+
+        client = smartcar.AuthClient(self.client_id, self.client_secret,
+                self.redirect_uri, self.scope, development=False)
+
+        actual = client.get_auth_url(force=True, state='stuff', single_select=single_select)
+
+        query = urlencode({
+            'response_type': 'code',
+            'client_id': self.client_id,
+            'redirect_uri': self.redirect_uri,
+            'approval_prompt': 'force',
+            'state': 'stuff',
+            'scope': ' '.join(self.scope),
```

Comment:
there should also be single_select here in the query right?

Reply:
yes!`